drakcore/drakcore/frontend/src/AnalysisLogs.js

```diff
@@ -0,0 +1,352 @@
+import React, { useState, useEffect, useRef } from "react";
+import OptionPicker from "./OptionPicker";
+import InfiniteLoader from "react-window-infinite-loader";
+import AutoSizer from "react-virtualized-auto-sizer";
+import { FixedSizeList as List } from "react-window";
+import { Tabs, TabItem } from "./Tabs.js";
+import api from "./api";
+
+function LogBrowser({
+  // Log entry component
+  children,
+  // Index of log file we want to browse
+  index,
+  // Function (from, to) => Promise<bytes>
+  queryData,
+}) {
+  // chunks object layout:
+  // {
+  //   0: [chunk 0 entries - objects],
+  //   1: [chunk 1 entries - objects],
+  //   ...
+  // }
+  const [chunks, setChunks] = useState(null);
+  const [lines, setLines] = useState(index.num_lines);
+  const pending = useRef(null);
+
+  // No index or only single marker => load only single chunk
+  const singleChunk = index == null || index.markers.length === 1;
+
+  // Load first chunk
+  useEffect(() => {
+    if (index !== null) {
+      setLines(index.num_lines);
+    }
+    queryData(0, singleChunk ? null : index.markers[1].offset - 1).then(
+      (entries) => {
+        if (singleChunk) {
+          setLines(entries.length);
+        }
+        setChunks({ 0: entries });
+      }
+    );
+  }, [singleChunk, index, queryData]);
+
+  useEffect(() => {
+    if (pending.current !== null) {
+      pending.current();
+      pending.current = null;
+    }
+  }, [chunks]);
+
+  if (chunks === null) {
+    return <>Loading...</>;
+  }
+
+  const findChunkIndex = (lineIndex) => {
+    if (singleChunk) {
+      return 0;
+    }
+
+    const markers = index.markers;
+    for (let i = 0; i < markers.length; i++) {
+      const chunk = markers[i];
+      if (chunk.line > lineIndex) {
+        return i - 1;
+      }
+    }
+    return markers.length - 1;
+  };
+
+  const isItemLoaded = (entry) => {
```

Comment:
Would rename as this doesn't load `Items` but `chunks`.

Reply:
Added comment in code